### PR TITLE
RunGroup geometry implementation, cleanup of hipo3/hipo4 utilities, add torus full map

### PR DESCRIPTION
**Pull request by raffaelladevita**
* Support to Run Group geometries via variations:
  * modified CTOF, FTOF, DC, EC, PCAL engines and geometry services to load geometry from variation 
  * modified CND, CTOF and CVT to implement detector z shift according to target position
  * update FTOF, EC, PCAL geometry services to use alignment tables (shifts only for JCSG, shift and rotations for ECAL in clas-geometry)
  * update magfield to shift solenoid according to target position as set from yaml/environment or ccdb via variation 
  * updated yaml files to provide usage examples
* Full, non-symmetric, torus map included in distribution, data processing yaml files updated accordingly
* Updated version of mon12 and kpp-plots
* Utilities renaming and clean-up 
* notsouseful-util renamed to recon-util
* hipo-utils for hipo4 and hipo3utils for hipo3
* decoder for hipo4 and decoder3 for hipo3

### CODE REVIEW

File: bin/env.sh [hunk 1]
```diff
@@ -10,6 +10,9 @@ DATAMINING=$DISTRO_DIR ; export DATAMINING
 if [ -z "$COAT_MAGFIELD_TORUSMAP" ]; then
     export COAT_MAGFIELD_TORUSMAP=Symm_torus_r2501_phi16_z251_24Apr2018.dat
 fi
+if [ -z "$COAT_MAGFIELD_TORUSSECONDARYMAP" ]; then
```
Comment (baltzell) [inline bin/env.sh]: Is this env var only used in the build script?  The code still only uses $COAT_MAGFIELD_TORUSMAP, right?

**Comment by raffaelladevita** [comment]
Yes, it’s defined in the env script  so that the build script can download this additional map but the primary map used in unit and advanced tests is set via $COAT_MAGFIELD_TORUSMAP

**Comment by baltzell** [comment]
So this is really just to get 2 maps in the tarball/release to make it easier for one of simulators or rungroup chefs to not have to manually download a map, and we don't really need an extra env var nor to download it in the build script?  But ok, it's just temporary until gemc supports the full map?

**Comment by raffaelladevita** [comment]
Yes, I think we can drop the secondary map as soon as we have a GEMC version that can handle the full map

**Pull request merged**